lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -157,12 +157,32 @@ def discover
         end
       end
 
-      node = @zk.get(@discovery['path'], :watch => true)
```

Comment:
This whole block of code got duplicated without need. Just use a variable to hold the path and this block can remain about the same.